hackathon_site/dashboard/frontend/src/components/teamDetail/ProjectDescription/ProjectDescription.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import { Formik, Form, Field, FormikValues } from "formik";
+import * as Yup from "yup";
+import { TextField, Button, Box, Grid } from "@material-ui/core";
+import { useDispatch } from "react-redux";
+import { updateProjectDescription } from "slices/event/teamDetailSlice";
+
+interface ProjectDescriptionProps {
+    teamCode: string;
+}
+
+const ProjectDescription = ({ teamCode }: ProjectDescriptionProps) => {
+    const dispatch = useDispatch();
+    const initialProjectDescription = "Write your project description here";
+    const [isEditing, setIsEditing] = useState(false);
+    const projectDescriptionSchema = Yup.object().shape({
+        projectDescription: Yup.string()
+            .max(500)
+            .required("Project description is required"),
+    });
+
+    const handleSubmit = async (values: FormikValues, { setSubmitting }: any) => {
+        try {
+            await dispatch(
+                updateProjectDescription({
+                    teamCode: teamCode,
+                    projectDescription: values.projectDescription,
+                })
+            );
+        } catch (error) {
```

Comment:
you should probably throw an error snackbar in case of an error, and also make a unit test for this snackbar

Reply:
actually is it necessary to have this try and catch block here? Doesn't the thunk already handle the asynchronous logic?

Reply:
Yeah, I believe it's not necessary